src/std/build-script.ss

```diff
@@ -61,7 +61,10 @@
                                      prefix: @prefix
                                      @build-spec)))
                             ([]
-                             (unless (file-exists? "build-deps")
+                             (unless (and (file-exists? "build-deps")
+                                          (file-newer? "build-deps" +this-source-file+)
+                                          (andmap (lambda (f) (file-newer? "build-deps" f))
+                                                  (buildspec-depfiles @build-spec)))
                                (displayln "... make deps")
```

Comment:
Happily, in the case of `build-deps` specifically, if `file-newer?` fails because we computed the deps less than a second after the last file was written, that's a self-correcting error, and next time it will be OK. This might warrant a comment in the source code, though.